data_structures/test_stack.cpp

```diff
@@ -1,59 +1,214 @@
 #include <iostream>
+#include <stdexcept>
+#include <vector>
 
-#include "./stack.h"
+#include "./stack.hpp"
+
+void assertTrue(const bool inValue) {
+    if (!inValue) {
+        std::cout << "Assertion fails!\n";
+        exit(1);
+    }
+}
```

Comment:
I don't understand the reasoning behind creating this when `assert` is already included in the file? `assert(a == b)` does the same thing so I don't understand the use case here, unless you're worried about `NDEBUG` which shouldn't be a concern here since we don't use that env variable (from what I can tell).

Reply:
I am concern about the `NDEBUG` and the [_standard_ `assert`](https://en.cppreference.com/w/cpp/error/assert). Tests should be able to fail no matter what kind of build is used and how the build is configured.

As [mentioned earlier](https://github.com/TheAlgorithms/C-Plus-Plus/pull/2235#issuecomment-1279699322): the perfect solution requires some proper testing framework.

Reply:
Your concern for the `NDEBUG` is valid but unnecessary I think - maybe @Panquesito7 can shed some light on this. The reason why I think it's unnecessary is that these `assert` statements are enough by themselves within this repositories build context. You say the tests should fail in all build scenarios but when `NDEBUG` is enabled the tests are not tests anymore and this repository would never add the `NDEBUG` macro to begin with so that shouldn't be something you should be concerned about.

Reply:
@tjgurwara99 based on the :+1: of @Panquesito7 I think you insist on using the standard assert. Now this is done. Just for information: the [CI](https://github.com/vil02/C-Plus-Plus/actions/runs/3856762403/jobs/6573363833) passes on my end.